### DISCUSSION TITLE
How to get ltex-ls to work with NvChad ?

### DISCUSSION THREAD

**Discussion by benzwt** [NvChad/NvChad#2308] [Q&A] [answered]
I failed to install ltex-ls to NvChad.
When I opened a md file, I got the following errors:

>[lspconfig] Cannot access configuration for ltex-ls. Ensure this server is listed in `server_configurations.md` or added as a custom server.

Could someone give me some advise ?
thanks!

Here are my steps for the installation of ltex-ls
1. add "ltex-ls" to mason in override.lua
2. :MasonInstallAll
3. make sure ltex-cli and ltex-ls are in  ~/.local/share/nvim/mason/bin/
4. append "ltex-ls" to the variable the local variable servers in custom/lspconfig.lua
5. add lspconfig.ltex.setup{} in custom/lspconfig.lua

my lua scripts are allow provided as below.
custom/configs/override.lua
```lua
M.mason = {
  ensure_installed = {
    "lua-language-server",
    "stylua",
    "ltex-ls",
    "clangd",
    "clang-format",
  },
}
```
configs/lspconfig.lua
``` lua
local on_attach = require("plugins.configs.lspconfig").on_attach
local capabilities = require("plugins.configs.lspconfig").capabilities

local lspconfig = require "lspconfig"

-- if you just want default config for the servers then put them in a table
local servers = { "rust_analyzer", "lua_ls", "clangd", "ltex-ls"}


for _, lsp in ipairs(servers) do
  lspconfig[lsp].setup {
    on_attach = on_attach,
    capabilities = capabilities,
  }
end

lspconfig.ltex.setup{}
```

**Comment by siduck** [comment, marked as answer]
mason package names dont represent actual lspserver names defined by lspconfig. please read https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#ltex

**Reply by benzwt** [reply]
I forgot to mention that, I did try to adding "ltex" to servers at custom/lspconfig.lua.
Finally, the Itex-ls is running.
The problem is I have to wait a few moment for the ltex-ls to be started.
I don't see animation of LSP (I'm using default) while the ltex-ls is being started.

**Reply by siduck** [reply]
@benzwt probably itex-ls doesnt provide certain stuff which is used by our statusline lsp progress module
